mllib/src/test/scala/org/apache/spark/ml/optim/aggregator/DifferentiableLossAggregatorSuite.scala

```diff
@@ -149,10 +152,43 @@ object DifferentiableLossAggregatorSuite {
 
     override def add(instance: Instance): TestAggregator = {
       val error = instance.label - BLAS.dot(coefficients, instance.features)
-      weightSum += instance.weight
-      lossSum += instance.weight * error * error / 2.0
-      (0 until dim).foreach { j =>
-        gradientSumArray(j) += instance.weight * error * instance.features(j)
+      val mathCls = Utils.classForName("java.lang.Math")
```

Comment:
@sarutak, I think we can just fix the test with tolerance e.g.) `assert(agg.weight ~== weightSum relTol 1e-5)` like the above one or https://github.com/apache/spark/blob/master/mllib/src/test/scala/org/apache/spark/ml/tuning/RandomRangesSuite.scala#L37

Up to my knowledge, this kind of precision diff is a common problem in ML side. cc @srowen, @WeichenXu123 @zhengruifeng FYI

Reply:
> I think we can just fix the test with tolerance at the assertion e.g.) assert(agg.weight ~== weightSum relTol 1e-5) like the above one

I considered using `~==` at first but can we accept the precision diff in this case?
The failed assertion uses `===` while some of other assertions use `~==` so I guessed `===` is intentionally used.
If we can use `~==` here, I'll fix with `~==`.

Reply:
I think its okay .. I did the similar things in few places before .. but should be good to have their input.

Reply:
O.K. Changed to use `~==`.

Reply:
An approximate equality test is appropriate here. I'm not sure why it varies in Java 11, but I could imagine tiny differences are possible. This is OK.

Reply:
> An approximate equality test is appropriate here. I'm not sure why it varies in Java 11, but I could imagine tiny differences are possible. This is OK.


+1